# 11.STATISTICS - 1- Assignment-1

# Problem Statement 1: 
You survey households in your area to find the average rent they are paying. Find the standard deviation from the following data:

$1550, $1700, $900, $850, $1000, $950.

In [6]:
from math import sqrt
def findMean(rentList):
    return sum(rentList)/len(rentList)

def findVariance(rentList):
    variance = 0.0
    mean = findMean(rentList)
    variance = sum((rent - mean)**2 for rent in rentList) / (len(rentList)-1)
    return variance

def findStandardDeviation(rentList):
    return findVariance(rentList)**0.5

rentList = [1550,1700,900,850,1000,950]


print ("Mean = " + str(findMean(rentList)))
print ("variance =" + str(findVariance(rentList)))
print("Standard Deviation =" + str(findStandardDeviation(rentList)))


Mean = 1158.3333333333333
variance =135416.6666666667
Standard Deviation =367.99003609699366


Output:
Mean = 1158.3333333333333
variance =135416.6666666667
Standard Deviation =367.99003609699366

# Problem Statement 2:
Find the variance for the following set of data representing trees in California (heights in
feet):
3, 21, 98, 203, 17, 9

In [7]:
treeHeightList = [3, 21, 98, 203, 17, 9]


print ("variance =" + str(findVariance(treeHeightList)))


variance =6219.9


Output:
variance =6219.9

# Problem Statement 3:
In a class on 100 students, 80 students passed in all subjects, 10 failed in one subject, 7
failed in two subjects and 3 failed in three subjects. Find the probability distribution of
the variable for number of subjects a student from the given class has failed in.

#Solution

For a random student, 

The probability of failing in 0 subjects, P(X=0) = 0.8

The probability of failing in 1 subjects, P(X=1) = 0.1

The probability of failing in 2 subjects, P(X=2) = 0.07

The probability of failing in 3 subjects, P(X=3) = 0.03

The probability distribution can be shown as:
    
    X      0,1,2,3
    P(X)   0.8, 0.1,0.07,0.03

